### PR TITLE
:bug: util.EnsureOwnerRef should ignore version when comparing objects

### PR DESCRIPTION
**Pull request by vincepri**
Signed-off-by: Vince Prignano <vincepri@vmware.com>

**What this PR does / why we need it**:
Updates our utilities to make sure we're properly updating the references instead of duplicating them if they actually refer to the same object with different versions.

/assign @ncdc @detiber

### CODE REVIEW

**Comment by ncdc** [comment]
Want to write any units?

File: util/util.go [hunk 1]
```diff
@@ -291,22 +290,44 @@ func MachineToInfrastructureMapFunc(gvk schema.GroupVersionKind) handler.ToReque
...
+	idx := FindOwnerRef(ownerReferences, ref)
+	if idx == -1 {
 		return append(ownerReferences, ref)
 	}
+	ownerReferences[idx] = ref
 	return ownerReferences
 }
 
+// FindOwnerRef returns the index of the owner reference in the slice if found, or -1.
+func FindOwnerRef(ownerReferences []metav1.OwnerReference, ref metav1.OwnerReference) int {
```
Comment (detiber) [inline util/util.go]: Might be good to rename this to make it clear it is returning an index rather than an OwnerRef.

Comment (detiber) [reply util/util.go]: Also, any reason to not make this private?

**Comment by vincepri** [comment]
@ncdc yeah I'll, wanted to make sure the logic is sound first

/hold

File: util/util.go (hunk 1 above)
Comment (vincepri) [reply util/util.go]: yeah sgtm

**Pull request merged**